packages/terra-functional-testing/src/reporters/file-output-reporter/index.js

```diff
@@ -0,0 +1,207 @@
+    this.hasResultsDir = this.hasResultsDir.bind(this);
+    this.setTestModule = this.setTestModule.bind(this);
+    this.printReport = this.printReport.bind(this);
+    this.getMessage = this.getMessage.bind(this);
+
+    this.setResultsDir();
+    this.hasResultsDir();
+  }
+
+  /**
+   * Sets results directory for the test run. Uses the wdio reporterOptions.outputDir if set, otherwise
+   * it outputs to tests/wdio/reports.
+   * @return null;
+   */
+  setResultsDir() {
+    this.resultsDir = path.join(process.cwd(), 'tests', 'wdio', 'reports');
+  }
+
+  /**
+   * Check and create reports dir if doesn't exist
+   * @return null
+   */
+  hasResultsDir() {
+    if (!fs.existsSync(this.resultsDir)) {
+      fs.mkdirSync(this.resultsDir, { recursive: true }, (err) => {
+        if (err) {
+          Logger.error(err.message, { context: LOG_CONTEXT });
+        }
+      });
+    }
+  }
+
+  /**
+   * Formatting the filename based on locale, theme, and formFactor
+   * @return null
+   */
+  fileNameCheck({ formFactor, locale, theme }, browserName, cid) {
+    const fileNameConf = ['fileOutput'];
+    if (locale) {
+      fileNameConf.push(locale);
+      this.resultJsonObject.locale = locale;
+    }
+
+    if (theme) {
+      fileNameConf.push(theme);
+      this.resultJsonObject.theme = theme;
+    }
+
+    if (formFactor) {
+      fileNameConf.push(formFactor);
+      this.resultJsonObject.formFactor = formFactor;
+    }
+
+    if (browserName) {
+      fileNameConf.push(browserName);
+    }
+
+    if (cid) {
+      fileNameConf.push(cid);
+    }
+
+    this.fileName = fileNameConf.join('-');
+  }
+
+  /**
+   * Set the package name to moduleName property if specsValue contains /package string
+   * @param {string} specsValue - File path of current spec file from runners
+   * @return null
+   */
+  setTestModule(specsValue) {
+    const index = specsValue.lastIndexOf(`packages${path.sep}`);
+    if (index > -1) {
+      const testFilePath = specsValue.substring(index).split(path.sep);
+      const moduleName = testFilePath && testFilePath[1]
+        ? testFilePath[1]
+        : process.cwd().split(path.sep).pop();
+      if (moduleName && moduleName !== this.moduleName) {
+        this.moduleName = moduleName;
+      }
+    }
+  }
+
+  onRunnerEnd(runner) {
+    this.runners.push(runner);
+    this.printReport(runner);
+  }
+
+  getMessage(runner) {
+    const results = this.getResultDisplay();
+    if (results.length === 0) {
+      return null;
+    }
+    const testLinks = runner.isMultiremote
+      ? Object.entries(runner.capabilities).map(
+        ([instanceName, capabilities]) => this.getTestLink({
+          config: { ...runner.config, ...{ capabilities } },
+          sessionId: capabilities.sessionId,
+          isMultiremote: runner.isMultiremote,
+          instanceName,
+        }),
+      )
+      : this.getTestLink(runner);
+    const output = [
+      ...this.getHeaderDisplay(runner),
+      '',
```

Comment:
What is this empty string used for?

Reply:
it is just used to format the data , I grabbed this functionality from the SpecReporter